pkg/process/util/orchestrator/pod.go

```diff
@@ -174,32 +174,42 @@ func extractPodMessage(p *v1.Pod) *model.Pod {
 	podModel.RestartCount = 0
 	for _, cs := range p.Status.ContainerStatuses {
 		podModel.RestartCount += cs.RestartCount
-		cStatus := model.ContainerStatus{
-			Name:         cs.Name,
-			ContainerID:  cs.ContainerID,
-			Ready:        cs.Ready,
-			RestartCount: cs.RestartCount,
-		}
-		// detecting the current state
-		if cs.State.Waiting != nil {
-			cStatus.State = "Waiting"
-			cStatus.Message = cs.State.Waiting.Reason + " " + cs.State.Waiting.Message
-		} else if cs.State.Running != nil {
-			cStatus.State = "Running"
-		} else if cs.State.Terminated != nil {
-			cStatus.State = "Terminated"
-			exitString := "(exit: " + strconv.Itoa(int(cs.State.Terminated.ExitCode)) + ")"
-			cStatus.Message = cs.State.Terminated.Reason + " " + cs.State.Terminated.Message + " " + exitString
-		}
+		cStatus := convertContainerStatus(cs)
 		podModel.ContainerStatuses = append(podModel.ContainerStatuses, &cStatus)
 	}
 
+	for _, cs := range p.Status.InitContainerStatuses {
+		podModel.RestartCount += cs.RestartCount
```

Comment:
Do we want to count the initContainers and their restarts as well?

Reply:
Building upon the discussion with Hassaim I will doublecheck how kubectl is doing it.

Reply:
After doublechecking it seems to be correct to add the restartCount to the overall restartCount as this models `kubectl` well.

See here:
```
vagrant@agent-dev-nam:~/go/src/github.com/DataDog/datadog-agent$ k describe pod init-demo

Init Containers:
  install:
    Container ID:  docker://b4062db0bf18b844eaded9c947754a741e7b39531a220024931d1f59960f307d
....
      Finished:     Wed, 24 Jun 2020 08:42:52 +0000
    Ready:          False
    Restart Count:  4
    
vagrant@agent-dev-nam:~/go/src/github.com/DataDog/datadog-agent$ k get pods
NAME                            READY   STATUS                   RESTARTS   AGE
init-demo                       0/1     Init:RunContainerError   5          3m12s
```

Reply:
excellent 👌